modules/ROOT/pages/sap/sap-s4hana-cloud-connector-upgrade-migrate.adoc

```diff
@@ -0,0 +1,75 @@
+= Upgrade SAP S/4HANA OData Cloud Connector to 2.0 - Mule 4
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Upgrade Anypoint Connector for SAP S/4HANA OData Cloud to take advantage of the increased security offered by using OAuth 2.0 authentication to connect to SAP S/4HANA OData. 
+
+== Supported Upgrade Paths
+
+This is direct upgrade from version 1.2.1 version to 2.0.0. 
+
+[%header,cols="50a,50a"]
+|===
+|From Version | To Version
+|1.2.1 |2.0.0
+|===
+
+== Changes in This Release
+
+* Support for OAuth 2.0 +
+** The *OAuth 2.0 Application Connection Provider* option has been added to *Connection* types. For information about the configuration parameters for this connection type, see xref:sap-s4hana-cloud-connector-reference.adoc[SAP S/4HANA OData Cloud Connector Reference]. 
+* Mule runtime engine (Mule) 4.2.1 or later is required. 
+* The default value for the *Service directory* field has been removed. 
+* The base package has been changed, which means that all exported classes now have different fully-qualified names. 
+* The group ID of the connector’s JAR has changed. 
+
+== Requirements and Limitations
+
+There are some requirements and limitations you should be aware of before you upgrade to the latest version.
+
+=== Requirements
+
+You must use Mule 4.2.1 or later. 
+
+=== Limitations
+
+Before you perform the upgrade, be aware that this upgrade breaks backward compatibility. 
+
+== Upgrade Prerequisites
+
+Before you perform the upgrade, you must create a backup of your files, data, and configurations in case you need to restore to the previous version. 
+
+
+== Upgrade Steps
+
+Follow these steps to perform the upgrade:
+
+. In Anypoint Studio, create a Mule project.
+. In the Mule Palette view, click *Search in Exchange*.
+. In the *Add Dependencies to Project* window, enter `SAP S/4HANA OData` in the search field.
+. In the *Available modules* section, select *SAP S/4HANA OData Cloud Connector* and click *Add*.
+. Click *Finish*.
+. Verify that the `mule-sap-s4hana-cloud-connector` dependency version is 2.0.0 in the `pom.xml` file in the Mule project.
+
+Studio upgrades the connector automatically.
+
+== Post Upgrade Steps
+
+After you install the latest version of the connector, follow these steps to verify the upgrade:
+
+. In Studio, verify that there are no errors in the *Problems* or *Console* views.
+. Check the project `pom.xml` file and verify that there are no problems.
+. Test the connection and verify that the operations work.
+
+== Troubleshooting
+
+If there are problems with caching the parameters and metadata, try restarting Studio.
+
+== Revert the Upgrade
+
+If it is necessary to revert to the previous version of SAP S/4HANA OData Cloud Connector, change the `mule-sap-s4hana-cloud-connector` dependency version 2.0.0 in the project’s `pom.xml` to the version to which you want to revert. 
```

Comment:
```suggestion
If it is necessary to revert to version 1.2.1, change the `mule-sap-s4hana-cloud-connector` dependency version 2.0.0 in the project’s `pom.xml` to the version to which you want to revert. 
```
Wait, I thought only 1.2.1 and 2.0.0 were involved in this whole process...?
Okay, so be sure that none of the suggested edits changes the meaning you intended....